pydantic/type_adapter.py

```diff
@@ -203,35 +211,75 @@ def __init__(
             )
 
-        config_wrapper = _config.ConfigWrapper(config)
-
-        core_schema: CoreSchema
+        self._type = type
+        self._config_wrapper = _config.ConfigWrapper(config)
+        self._parent_depth = _parent_depth
+        if module is None:
+            f = sys._getframe(1)
+            self._module_name = cast(str, f.f_globals.get('__name__', ''))
+        else:
+            self._module_name = module
+
+        self._schema_initialized = False
+        if not self._defer_build(type, config):
+            # Immediately initialize the core schema, validator and serializer
+            _, _, _ = (self.core_schema, self.validator, self.serializer)
+
+    @cached_property
+    def core_schema(self) -> CoreSchema:
+        """Core schema"""
         try:
-            core_schema = _getattr_no_parents(type, '__pydantic_core_schema__')
+            return _getattr_no_parents(self._type, '__pydantic_core_schema__')
         except AttributeError:
-            core_schema = _get_schema(type, config_wrapper, parent_depth=_parent_depth + 1)
+            return _get_schema(self._type, self._config_wrapper, parent_depth=self._parent_depth + 3)
+        finally:
+            self._schema_initialized = True
 
-        core_config = config_wrapper.core_config(None)
-        validator: SchemaValidator
+    @cached_property
+    def validator(self) -> SchemaValidator:
+        """Validator"""
         try:
-            validator = _getattr_no_parents(type, '__pydantic_validator__')
+            return _getattr_no_parents(self._type, '__pydantic_validator__')
         except AttributeError:
-            if module is None:
-                f = sys._getframe(1)
-                module = cast(str, f.f_globals.get('__name__', ''))
-            validator = create_schema_validator(
-                core_schema, type, module, str(type), 'TypeAdapter', core_config, config_wrapper.plugin_settings
-            )  # type: ignore
-
-        serializer: SchemaSerializer
+            return create_schema_validator(
+                schema=self.core_schema,
+                schema_type=self._type,
+                schema_type_module=self._module_name,
+                schema_type_name=str(self._type),
+                schema_kind='TypeAdapter',
+                config=self._core_config,
+                plugin_settings=self._config_wrapper.plugin_settings,
+            )
+
+    @cached_property
+    def serializer(self) -> SchemaSerializer:
+        """Serializer"""
         try:
-            serializer = _getattr_no_parents(type, '__pydantic_serializer__')
+            return _getattr_no_parents(self._type, '__pydantic_serializer__')
         except AttributeError:
-            serializer = SchemaSerializer(core_schema, core_config)
+            return SchemaSerializer(self.core_schema, self._core_config)
+
+    @classmethod
+    def _defer_build(cls, type_: Any, type_adapter_config: ConfigDict | None) -> bool:
```

Comment:
Why not use `self._type` and self._config_wrapper`?

Reply:
Done, changed to instance funcs